codalab/worker/file_util.py

```diff
@@ -135,8 +146,7 @@ def zip_directory(
         try:
             proc = subprocess.Popen(args, stdout=subprocess.PIPE, cwd=directory_path)
             proc.wait()
-            with open(tmp_zip_name, "rb") as out:
-                return BytesIO(out.read())
+            return open(tmp_zip_name, "rb")
```

Comment:
Was there a reason that we needed to read first and not just do the simpler thing you're doing?

Reply:
Seems like these subtle changes can have a large performance effect on the system for large files, just want to understand a bit better.

Reply:
No, I don't think there was a good reason for doing that earlier. However, it looks like `zip_directory` is not actually used in the codalab codebase (I think I had added it initially when we were using .zip files for Blob Storage -- but it's just used in this test now: https://github.com/codalab/codalab-worksheets/blob/43eaddc8013d484fdfdde91eb07dc6eaf7902601/tests/unit/worker/file_util_test.py#L138).